### ISSUE TITLE
When using a connection with an external app, results in opening the same external app continuesly

### ISSUE THREAD

**Issue by MaRRiK74** [mRemoteNG/mRemoteNG#325]
![image](https://cloud.githubusercontent.com/assets/19833001/21542121/e8b5ff40-cdbb-11e6-9ded-df4100b1e6d4.png)

|Operating system | Windows 2016 x64 |
|mRemoteNG version| 1.75.6179.28241  |

Log content (snip):

2016-12-29 11:04:18,486 [1] ERROR- ExternalApp.Start() failed.
The operation was canceled by the user
2016-12-29 11:04:18,486 [1] ERROR- Killing Int App Process failed!
Object reference not set to an instance of an object.
2016-12-29 11:04:18,486 [1] ERROR- Dispose of Int App process failed!
Object reference not set to an instance of an object.
2016-12-29 11:04:18,502 [1] ERROR- Killing Int App Process failed!
Object reference not set to an instance of an object.
2016-12-29 11:04:18,502 [1] ERROR- Dispose of Int App process failed!
Object reference not set to an instance of an object.
2016-12-29 11:04:18,502 [1] ERROR- ExternalApp.Start() failed.
The operation was canceled by the user

Cannot remove the specified item because it was not found in the specified Collection.
2016-12-29 11:04:18,627 [32] INFO - Connection Event Closed
2016-12-29 11:04:18,627 [32] INFO - Connection to  via IntApp closed by user ******.
2016-12-29 11:04:18,627 [31] INFO - Connection to  via IntApp closed by user ******.
2016-12-29 11:04:18,643 [1] ERROR- Couldn't close tab

**Comment by kmscode** [comment]
Could you provide some more details about the external app being used?

**Comment by MaRRiK74** [comment]
It happens with multiple external apps like mmc.exe, .ica (citrix file), sql management studio, etc. This issue started since 1.74 beta3. Never had it before.

**Comment by MaRRiK74** [comment]
Just did a quick test. make D;\test.txt as an external app, added it as a new connection. When opening the connection notepad will be opened about 100 times:

![image](https://cloud.githubusercontent.com/assets/19833001/21680454/28b7764e-d34a-11e6-90af-2c88fae8b6e1.png)

**Comment by kmscode** [comment]
The logs provided in your original comment (Killing/Dispose Int App Process failed!)relate to using an external tool that is (attempted) to be integrated into mremoteng... However, the TEST2 configuration you provided doesn't attempt to integrate...

I tested the later scenario and didn't have any problems.

Regardless, I did add in some additional protections for a problem that I ran into when my external tool configuration didn't exist properly. I also added in some better logging to help diagnose this further should none of of the other changes help the problem that you're running into... These will be included in the next beta build within the next few days/weeks (we haven't discussed the exact timeframe yet).

**Comment by MaRRiK74** [comment]
I will test the new release as soon as it will be available.

FYI..
In the meanwhile I have tested the issue with a fresh install/download (portable) of the current beta on Windows Server 2008 R2, Windows Server 2016 and Windows 10 (all x64) and the problem exists on all machines.

**Comment by kmscode** [comment]
99.9999% sure that I've fixed this. I was ultimately able to reproduce the errors listed in your original log.

In short, multiple calls to .Close() were happening. This is why we see duplicate/repeating messages. In addition, as part of the .Close() calls, we were attempting to go though cleanup code that is intended only for Integrated External apps. 

Prevented multiple .Close() calls and prevented getting into unnecessary code. Added some more clear log messages as well.
